### PR TITLE
K8SPSMDB-380 Mount users secret to mongo pod

### PR DESCRIPTION
**Pull request by nonemax**
[K8SPSMDB-380](https://jira.percona.com/browse/K8SPSMDB-380) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

We need to restart mongod pods if we change monitor user because we use it for liveness checks. So in this PR we mount users secret into the pod for catching on a fly users pass change

### CODE REVIEW

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | failed |
| scheduled-backup | failed |
| operator-self-healing | passed |
| security-context | failed |
| demand-backup-sharded | passed |
| upgrade | failed |
| service-per-pod | failed |
| rs-shard-migration | passed |
| liveness | passed |
| smart-update | passed |
| version-service | failed |
| users | passed |

File: healthcheck/tools/db/config.go [hunk 1]
```diff
@@ -77,10 +79,18 @@ func NewConfig(app *kingpin.Application, envUser string, envPassword string) *Co
 		"username",
 		"mongodb auth username, this flag or env var "+envUser+" is required",
 	).Envar(envUser).Required().StringVar(&db.DialInfo.Username)
-	app.Flag(
-		"password",
-		"mongodb auth password, this flag or env var "+envPassword+" is required",
-	).Envar(envPassword).Required().StringVar(&db.DialInfo.Password)
+	pass, err := ioutil.ReadFile("/etc/users-secret/MONGODB_CLUSTER_MONITOR_PASSWORD")
+	if err != nil && err != os.ErrNotExist {
+		return nil, fmt.Errorf("read MONGODB_CLUSTER_MONITOR_PASSWORD: %v", err)
```
Comment (blez) [inline healthcheck/tools/db/config.go:84]: let's user `errors` package instead of fmt

**Review by blez — APPROVED** [review]

File: healthcheck/tools/db/config.go (hunk 1 above)
Comment (nonemax) [reply healthcheck/tools/db/config.go]: https://github.com/percona/percona-server-mongodb-operator/pull/551/commits/901f5a3d5126c2724aa55c310b3bf0a791c18ecc

**Review by blez — DISMISSED** [review]

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | failed |
| demand-backup-sharded | failed |
| rs-shard-migration | passed |
| upgrade | passed |
| liveness | passed |
| upgrade-sharded | passed |
| smart-update | passed |
| version-service | failed |
| users | passed |
| data-sharded | failed |

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | failed |
| service-per-pod | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | failed |
| users | passed |
| data-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| self-healing | passed |
| scaling | passed |
| scheduled-backup | failed |
| arbiter | passed |
| security-context | failed |
| operator-self-healing | passed |
| demand-backup-sharded | passed |
| service-per-pod | failed |
| upgrade | failed |
| rs-shard-migration | failed |
| liveness | passed |
| smart-update | passed |
| version-service | failed |
| users | passed |
| upgrade-sharded | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| arbiter | passed |
| init-deploy | passed |
| operator-self-healing | passed |
| service-per-pod | failed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | failed |
| users | passed |
| data-sharded | passed |

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-551-4b9e729c

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| arbiter | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| upgrade | passed |
| version-service | passed |
| upgrade-sharded | passed |
| users | passed |
| data-sharded | passed |

**Pull request merged**